code/game/machinery/doors/airlock.dm

```diff
@@ -1006,9 +965,8 @@
 		return
 	if(hasPower())
 		if(forced)
-			if(isElectrified())
-				shock(user,100)//it's like sticking a forck in a power socket
-				return
+			if(SEND_SIGNAL(src, COMSIG_AIRLOCK_ATTACKBY_POWERCROWBAR, user) & COMSIG_ELECTRIFIED_SHOCK)
```

Comment:
This signal should be sent regardless of state of the airlock, let the component decide if they get shocked